File: 2971-find-polygon-with-the-largest-perimeter/2971-find-polygon-with-the-largest-perimeter.py
# Time complexity: O(N log N)
# Space complexity: O(1)
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        l: int = -1
        s: int = sum(nums[:2])
        for i in range(2, len(nums)):
            if nums[i] < s:
                l = nums[i] + s
            s += nums[i]
        return l

File: 2971-find-polygon-with-the-largest-perimeter/2971-find-polygon-with-the-largest-perimeter.java
class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        long sum = 0;
        for(int n : nums) {
            sum += n;
        }

        for (int i = nums.length - 1; i >= 2; i--) {
            if (sum - nums[i] > nums[i]) {
                return sum;
            } else {
                sum -= nums[i];
            }
        }
        return -1;
    }
}